src/java.base/share/classes/jdk/internal/foreign/layout/ValueLayouts.java

```diff
@@ -312,7 +319,7 @@ OfDoubleImpl dup(ByteOrder order, long byteAlignment, Optional<String> name) {
         }
 
         public static OfDouble of(ByteOrder order) {
-            return new OfDoubleImpl(order, ADDRESS_SIZE_BYTES, Optional.empty());
+            return new OfDoubleImpl(order, Utils.IS_AIX ? 32 : ADDRESS_SIZE_BYTES, Optional.empty());
```

Comment:
The second argument is expressed in bytes: `32` should be `4`.

Reply:
This class should also implement:
```java
        @Override
        public boolean hasNaturalAlignment() {
            return Utils.IS_AIX ? (byteAlignment() == 4) : super.hasNaturalAlignment();
        }
```

Reply:
Fixed and added `hasNaturalAlignment()`.